src/components/Card/CardBox.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { useState } from "react";
+import Card from "./Card";
+
+const CardBox: React.FC = () => {
+  const [isCardFrontActive, setIsCardFrontActive] = useState(true);
+  const [isCardBackActive, setIsCardBackActive] = useState(false);
+  const [content, setContent] = useState({
+    frontText: "front text",
+    backText: "back text",
+  });
+  const [isEditFrontActive, setIsEditFrontActive] = useState(false);
+  const [isEditBackActive, setIsEditBackActive] = useState(false);
+  const text = isCardBackActive ? content.backText : content.frontText;
+  const handleEditButton = (type:string) => {
+    if (type === "front") {
+        setIsCardFrontActive(false)
+        setIsEditFrontActive(true)
+    }
+    else {
+        setIsCardBackActive(false)
+        setIsEditBackActive(true)
+    }
+  }
+
+  return (
+    <>
+      {((isCardFrontActive && !isCardBackActive) || (isEditFrontActive && !isEditBackActive)) && (
+        <Card text={text} type={isEditFrontActive ? "editfront" : "front"} />
```

Comment:
**nitpick:** This got really complicated - you could just differ active side by `activeCardSide`, and then just pass `editing`